dracut/30afterburn/afterburn-network-kargs.service

```diff
@@ -0,0 +1,18 @@
+[Unit]
+Description=Afterburn Initrd Setup Network Kernel Arguments
+
+# This service may produce additional kargs fragments,
+# which are then consumed by dracut-cmdline(8).
+DefaultDependencies=no
+Before=dracut-cmdline.service
+Before=ignition-fetch.service
+After=systemd-journald.socket
+
+OnFailure=emergency.target
+OnFailureJobMode=isolate
+
+[Service]
+Environment=AFTERBURN_OPT_DEFAULT_VALUE=--default-value=ip=dhcp,dhcp6
```

Comment:
The kernel networking defaults are critically important value, and this feels like a rather obscure place for them to live.
How about we add them to fedora-coreos-config in e.g. `/usr/share/coreos-default-network-kargs` or something, and then here do:
`EnvironmentFile=/usr/share/coreos-default-network-kargs`
?

Reply:
I agree.
Though, this introduces an hard-requirement on a file which only lives in some -config repo.
Should I keep this current line and just add the `EnvironmentFile=-` afterwards, so that the explicit entry under -config is only *overriding* it if it exists?

Reply:
But then we have (potentially) two versions; we'd silently use this one if the other one happened to not exist or we typo'd a path.

Perhaps a build-time option is best?  Would help enable other distributions.

Reply:
Yes, that's how overriding works by design.
There is also an in-code default [here](https://github.com/coreos/afterburn/pull/390/files#diff-2f2a33f6cbc0b21a19815f89505a9f8cR47) (which can be turned into a build-time option, yes, but I think the unit override is more useful and with lower complexity).

Unless you are actually thinking of making this a required parameter and dropping all the defaults from this repo. In which case, I can rework some bits so that the unit will actually fail if `--default-value` is missing, and inject that **only** via a -config dropin.
Is this UX closer to the flow you had in mind? That is, your preference is for a single definition with hard-fail, not multiple levels of overrides?

Reply:
> I can rework some bits so that the unit will actually fail if --default-value is missing, and inject that only via a -config dropin.
> Is this UX closer to the flow you had in mind? That is, your preference is for a single definition with hard-fail, not multiple levels of overrides?

I think so yes.  This feels like a global property of the operating system that I wouldn't expect to be hardcoded in afterburn.

Think about how many gyrations we went through with the switch from `dhcp` to `dhcp,dhcp6`, having that be revs to `afterburn.rpm` just feels weird.

So yes, my proposal is that this is a file in some format (text, env foo=bar, JSON, whatever) that we ship in fedora-coreos-config.

(Although, complicating things is that it needs to end up in the initramfs so we'd need a new dracut "module" just to copy the file, but eh.)

This all said...I'm not opposed to landing this as is either to unblock further work and iterating on this as a followup.

Reply:
No worries, I'd rather get the integration/design part right upfront. I've pushed a fixup for the flow above. Thanks for the feedback!